### PR TITLE
Add installer hooks 

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

1. Add installer hooks 
2. hook execution order is alphabetical by name.
3. hooks that are not registered in sealer will return an error, at present ,only add shell hook at installer hook list.
4.  if "on" field at hookConfig is not set, then use all host ip as default.
5.  Transfer plugin "action" to hook "phase".
6. complete installation process to set RuntimeConfig.

### CODE REVIEW

File: pkg/cluster-runtime/hook.go [hunk 1]
```diff
@@ -1,51 +1,152 @@
...
+	On On `json:"on,omitempty"`
+}
+
+type HookConfigList []HookConfig
+
+func (r HookConfigList) Len() int           { return len(r) }
+func (r HookConfigList) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
+func (r HookConfigList) Less(i, j int) bool { return r[i].Name < r[j].Name }
+
+func (i *Installer) AddHook(phase Phase, hook HookConfig) error {
```
Comment (VinceCui) [inline pkg/cluster-runtime/hook.go:88]: where use AddHook?

File: pkg/cluster-runtime/hook.go [hunk 2]
```diff
@@ -1,51 +1,152 @@
...
 
 	i.hooks[phase] = append(i.hooks[phase], hook)
 
 	return nil
 }
 
-// run hooks
+// run hooks by Phase.
 func (i *Installer) runHook(phase Phase) error {
+	hookConfigList, ok := i.hooks[phase]
```
[the remark this answers is not in the document]
Comment (kakaZhou719) [reply pkg/cluster-runtime/hook.go:100]: got you

File: pkg/cluster-runtime/hook.go [hunk 3]
```diff
@@ -1,51 +1,152 @@
...
+	}
+
+	// sorted by hookConfig name in alphabetical order
+	sort.Sort(hookConfigList)
+
+	for _, hookConfig := range hookConfigList {
+		hosts := i.infraDriver.GetHostIPListByRole(string(hookConfig.On))
+
+		// if "on" field is not set, then use all host ip as default.
+		if len(hosts)==0{
```
[the remark this answers is not in the document]
Comment (VinceCui) [reply pkg/cluster-runtime/hook.go:112]: better

File: pkg/cluster-runtime/hook.go [hunk 4]
```diff
@@ -1,51 +1,157 @@
...
+	hookConfigList, ok := i.hooks[phase]
+	if !ok {
+		return fmt.Errorf("failed to load hook at phase: %s", phase)
 	}
 
-	i.hooks[phase] = append(i.hooks[phase], hook)
+	// sorted by hookConfig name in alphabetical order
+	sort.Sort(hookConfigList)
+	for _, hookConfig := range hookConfigList {
+		targetHosts = i.infraDriver.GetHostIPListByRole(string(hookConfig.Scope))
```
Comment (VinceCui) [inline pkg/cluster-runtime/hook.go:95]: still bug

**Comment by VinceCui** [comment]
LGTM

**Pull request merged**